v2/internal/test/fakeworkloadapi/workload_api.go

```diff
@@ -128,6 +127,7 @@ func (w *WorkloadAPI) SetJWTBundles(jwtBundles ...*jwtbundle.Bundle) {
 }
 
 type workloadAPIWrapper struct {
+	workload.UnimplementedSpiffeWorkloadAPIServer
```

Comment:
Sorry for the noob question here, but could you help me understand the nature of this addition?

Reply:
The `Unimplemented*Server` type is a struct that implements all of the methods of the interface and returns NOT_IMPLEMENTED when they are invoked. It is a method by which you can maintain forwards compatability with existing server implementations when new RPCs are added. There is a huge debate on whether or not it is actually helpful, since with it embedded, the compiler can no longer let you know if you have not implemented all the service methods. In any case, with the newly generated code, _by default_, your service implementations have to embed either 1) `Unimplemented*Server` or `Unsafe*Server`. The `Unsafe*Server` is an explicit opt-out of the forward compatibility guarantees. You can also pass a flag to the protoc-gen-go-grpc to opt out of the requirement to embed one or the other.

Personally, I think the Unsafe was a poor name choice. There is no benefit here from a forward compatibility standpoint at the protocol layer, just whether or not your build breaks. "Unsafe" seems like a strong word for that kind of breakage.

In this specific instance, I waffled between Unimplemented and Unsafe..... i thought it was mostly sixes for this fake implementation. I could be (trivially) convinced to go with Unsafe.

Reply:
Thanks for the explanation @azdagron 

> i thought it was mostly sixes for this fake implementation. 

Yea, sounds like it. I'm fine either way 👍